tracer/test/Datadog.Trace.TestHelpers.AutoInstrumentation/AspNetCoreTestFixture.cs

```diff
@@ -259,6 +258,27 @@ private async Task EnsureServerStarted(bool sendHealthCheck)
             }
         }
 
+        private bool IsPortListening(int port)
+        {
+            try
+            {
+                using (var client = new TcpClient())
+                {
+                    var task = client.ConnectAsync("127.0.0.1", port);
+                    if (task.Wait(1000))
```

Comment:
_Theoretically_ we could end up waiting 1s here, and then we wait for the interval again in the outer loop, so we technically _could_ end up waiting for longer than `maxMillisecondsToWait`, but I don't think that's a big problem.

Reply:
Good catch! That would also apply to the sendHealthCheck case. I have updated the code to minimize the impact of this.